### DISCUSSION TITLE
Can I start Kratos without courier?

### DISCUSSION THREAD

**Discussion by kwohlfahrt** [ory/kratos#3633] [Q&A]
Is it possible to disable the `courier` component entirely, if I don't intend to use (or collect) user email addresses? I'm aware that this will mean a number of features (e.g. account recovery) aren't available.

If I simply omit the `courier` section of the configuration, I get the error message below:

```
The configuration contains values or keys which are invalid:
courier.smtp.connection_uri: <nil>
                             ^-- one or more required properties are missing

time=2023-11-19T20:49:27Z level=error msg=Unable to instantiate configuration. audience=application error=map[message:I[#/courier/smtp] S[#/properties/courier/properties/smtp/required] missing properties: "connection_uri"] service_name=Ory Kratos service_version=v1.0.0
Error: I[#/courier/smtp] S[#/properties/courier/properties/smtp/required] missing properties: "connection_uri"
```

**Comment by vinckr** [comment]
Hello @kwohlfahrt 
Did you find a solution for you use case? 
I don't think disabling courier is supported without some modifications.
If you don't intend to use or collect user email addresses, you could provide a dummy SMTP server configuration to bypass this requirement. However, this is not recommended as it might lead to unexpected behavior in the system.
Also as you pointed out it disables recovery entirely, which seems suboptimal.

**Comment by OskarsPakers** [comment]
I think it would make sense to run kratos without courier and smtp.
We have use case when there is no password login and only social provider login and user schema without email. It does not make sense to enable courier and recovery flow. Does it?

**Reply by vinckr** [reply]
Yes, I agree in that case you wont need the courier. 
I am not sure however what the best way to run Ory Kratos without it is, since I have no experience with that use case personally. 

However after digging in community discussions a bit, I found this recommendation

>set the courier to something and then don't set it up.

>It just needs to be a URL starting with smtp:// or smtps:// though. So maybe you can just set it a “dummy” URL, something like smtps://localhost:1025 for example? If you don’t trigger any mails kratos shouldn’t try to connect to it. Keep in mind though, that a lot of features of kratos currently rely on emails. So without it those features will not work.

Since you wont use it anyway this seems like a reasonable fix to me. 

WDYT @kwohlfahrt @OskarsPakers ?

**Reply by OskarsPakers** [reply]
Yes, that is currently how we configure it. Helm chart also has option courier.enabled=false which does not deploy courier. This is ok workaround with only drawback that the configuration is not clean and you need to specify the dummy value.
Cleaner solution would be to drop requirement of the configuration value if in practice, kratos can work without courier.
